src/backend/distributed/README.md

```diff
@@ -0,0 +1,2079 @@
+
+Hash-distributed tables can be **co-located** with each other, such that the shards with the same hash value range are always on the same node. From the planner point-of-view, range-distributed tables can also be colocated, but shards are not managed by Citus.  
+
+Shards can be replicated (deprecated), in which case they have multiple shard placements. Writes to the placements are performed using 2PC and use aggressive locking to avoid diverging. 
+
+**Reference tables** are created using SELECT create_reference_table(..). They have a single shard, which is replicated to all nodes. It is possible to have a node without reference table replicas, but in that case the reference tables are replicated before the next rebalance. Reference tables are always co-located with each other and have a common co-location ID. 
+
+Writes to a reference table are performed using 2PC and use aggressive locking to avoid diverging. 
+
+**Single shard tables** are a special type of distributed table without a distribution column and with a single shard. When using schema-based sharding, tables created in a distributed schema automatically become single shard tables. Single shard tables can be co-located with each other, but not replicated. 
+
+**Citus local tables**: A single shard table that can only be placed on the coordinator and are primarily used as a drop-in replacement for regular PostgreSQL tables when creating foreign keys to/from reference tables. All Citus local tables are implicitly co-located with each other, but do not have a co-location ID. 
+
+All Citus table types have the notion of a “shard”, though in many cases there is only a single shard.  
+
+**Shard**: The table that contains the actual data in a Citus table. Shards reside in the same schema as the Citus table and are named after the Citus table with the shard ID as a suffix. Shards are hidden from catalog queries unless you SET citus.override_table_visibility TO off. Hash-distributed tables have multiple shards which each have a distinct shardminvalues/shardmaxvalue. 
+
+**Colocation group**: A set of distributed tables that have the same co-location ID, such that their shards are always co-located. 
+
+**Shard group**: A set of shards with the same shardminvalue/shardmaxvalue that are part of the same co-location group. Citus guarantees that shards from the same shard group are always placed on the same node group. 
+
+**Shard placement**: The assignment of a shard to a node group. There can be multiple placements of the same shard if the table is replicated (e.g. reference tables). 
+
+**Shard placement group**: The assignment of a shard to a node group must be the same for all shards in a shard group, since those are always co-located. We’ll refer to the group of placements of a shard group as a shard placement group. A single shard group can have multiple shard placement groups. 
```

Comment:
I found the distinction between shard placement and shard placement group to be very subtle, hard to understand.

Am I understanding correctly that individual shards have a placement row (multiple for replication) in a metadata table. The shard placement group is just an abstract way of referring to all the placements of shards in a shard group. "Multiple shard placement groups" therefore means that many different combinations of placements?